akka-stream/src/main/scala/akka/stream/scaladsl/OperationAttributes.scala

```diff
@@ -57,6 +63,7 @@ object OperationAttributes {
   final case class InputBuffer(initial: Int, max: Int) extends Attribute
   final case class Dispatcher(dispatcher: String) extends Attribute
   final case class SupervisionStrategy(decider: Supervision.Decider) extends Attribute
+  final case class MaterializationName(n: String) extends Attribute
```

Comment:
I agree, it would be better if we could pass this in some other way. Didn't Roland add the materializer to some context for the async stage? Perhaps that approach can be extended?